typing/typetexp.ml

```diff
@@ -1007,6 +1017,15 @@ let report_error env ppf = function
       Printtyp.reset_and_mark_loops ty;
       fprintf ppf "@[The type %a@ is not an object type@]"
         Printtyp.type_expr ty
+  | Unbound_value_missing_rec (lid, loc) ->
+      fprintf ppf
+        "Unbound value %a" longident lid;
+      spellcheck ppf fold_values env lid;
+      let (_, line, _) = Location.get_pos_info loc.Location.loc_start in
+      fprintf ppf
+        "@.@[%s %i@]"
+        "Hint: If this is a recursive definition, you should add a 'rec' keyword on line"
```

Comment:
This line seems to be a bit long, and the resulting
user-facing message is larger than what is usually
output by the `Format` module. It might be worthwhile
to both split the line and insert a breakable space.

Reply:
See also `Format.pp_print_text : formatter -> string -> unit` which turns all spaces into breakable spaces; here it could be used by turning `%s` into`%a`.

Reply:
(Actually, I am not certain we want to blindly turn all spaces
into breakable ones. Given the shape of the message, there
is a possibility that the break would occur right before the line
number, which is not very nice.)

Reply:
Also, I am wondering if it would be possible to avoid "@." by opening a vectical box : 
```OCaml
fprintf ppf
        "@[<v>@[Unbound value %a@]" longident lid;
      spellcheck ppf fold_values env lid;
      let (_, line, _) = Location.get_pos_info loc.Location.loc_start in
      fprintf ppf
        "@,@[%s %i@]@]"
        "Hint: If this is a recursive definition, you should add a 'rec' keyword on line"
```
since the behavior of `@.` tends to mask problems with the formatting.

Reply:
@gasche sure, but we can get a break before the word "line";
in the surrounding code, breakable spaces are explicitly placed
at strategic points.